lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,7 +154,10 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
-        evt_time = time()
+        if user.is_anonymous():
+            # Anonymous users cannot be persisted to the database, so let's just use
+            # what we have.
+            return
```

Comment:
My take is that it's not presently hurting anything, and it's giving us useful information that will inform future decision making. CSM remains one of our biggest long term scaling concerns, and I'm not convinced that the current system is going to scale smoothly for that long. We've seen occasional write latency spikes that I can't really explain very well. I'll send you the link to some details on one of those.

In any case, unless it is actively causing us trouble, I would very much appreciate it if we could keep this instrumentation in.